### PR TITLE
Use spago as build tool / package manager

### PR DESCRIPTION
**Pull request by sharkdp**
FYI @mhmdanas

### CODE REVIEW

**Comment by sharkdp** [comment]
This seems to work just fine :-)

File: README.md [hunk 1]
```diff
@@ -42,10 +42,11 @@ Calculate the oscillation period *T = 2π sqrt(L/g)* of a pendulum with length *
 "0.89714 s"
 ```
 
-## Installation
+## Development
 ```
-bower install purescript-quantities
-npm install decimal.js
+npm install
+spago build
```
Comment (triallax) [inline README.md]: I'm not sure, but I believe that `spago test` already builds modules. Could you please check?

File: packages.dhall [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          ]
+      , repo =
+          "https://github.com/hdgarrood/purescript-benchotron.git"
+      , version =
+          "v7.0.0"
+      }
+-------------------------------
+-}
+let upstream =
+      https://github.com/purescript/package-sets/releases/download/psc-0.14.0-20210409/packages.dhall sha256:e81c2f2ce790c0e0d79869d22f7a37d16caeb5bd81cfda71d46c58f6199fd33f
```
Comment (triallax) [inline packages.dhall]: Perhaps update the set by running `spago upgrade-set`.

File: README.md (hunk 1 above)
Comment (sharkdp) [reply README.md]: Right. removing the `spago build` line...

File: packages.dhall (hunk 2 above)
Comment (sharkdp) [reply packages.dhall]: Running this command does not seem to have any effect? ~~I guess this is the latest version?~~ (no, it's not)

Comment (triallax) [reply packages.dhall]: Hmm, weird. Can you try `spago upgrade-set --tag "psc-0.14.2-20210629"`?

Comment (sharkdp) [reply packages.dhall]: Looks like my PureScript version is too old for that package release. Need to update that first.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build source
+        run: spago build
+
```
Comment (triallax) [inline .github/workflows/ci.yml]: Probably remove this too:
```suggestion
```

Comment (sharkdp) [reply .github/workflows/ci.yml]: :+1:

**Comment by triallax** [comment]
You presumably also want to follow the instructions in https://github.com/purescript/spago#upgrade-the-package-set so you can actually publish the library.

**Comment by sharkdp** [comment]
> You presumably also want to follow the instructions in https://github.com/purescript/spago#upgrade-the-package-set so you can actually publish the library.

Which ones exactly? We are on the latest package set now, right?

**Comment by triallax** [comment]
Ah, sorry, wrong link. Here's the right one: https://github.com/purescript/spago#publish-my-library

**Comment by sharkdp** [comment]
I added a `license` and `repository` field. Do we need a new release right now?

**Comment by triallax** [comment]
The PR looks good to me, so yes, you can merge it and then try to publish a new release (or maybe do this before merging, up to you). Tell me if anything goes wrong.

**Comment by sharkdp** [comment]
Thank you very much for the review and the help!

**Comment by sharkdp** [comment]
> publish a new release

Done

**Comment by triallax** [comment]
I just noticed that the test dependencies are included in the generated bower.json, which is not ideal. https://github.com/purescript/spago#devdependencies-testdependencies-or-in-general-a-situation-with-many-configurations shows how to have a separate test configuration. You've already published a new release, but how about doing this?

**Comment by sharkdp** [comment]
Thank you. I did something that I normally wouldn't do on a larger project. I reset the master branch to before the release and removed the tag. I hadn't run `pulp publish` yet anyways.

**Comment by triallax** [comment]
Yeah, I think it's appropriate in this case. I can make a PR soon (tomorrow?) and do this, or you can do it yourself if you feel like it.

**Comment by sharkdp** [comment]
> I can make a PR soon (tomorrow?) and do this

That would be great. No rush please!

**Pull request merged**